src/config.rs

```diff
@@ -10,42 +10,34 @@ use std::convert::TryFrom;
 
 #[derive(Deserialize, Debug)]
 struct RawConfig {
-    internal_url: String,
+    #[serde(flatten)]
+    raw_base_config: RawBaseConfig,
     phonenumber: String,
 
     dtmf_length: usize,
     result_length: usize,
-
-    decryption_privkey: EncryptionKeyConfig,
-    signature_pubkey: SignKeyConfig,
 }
 
 #[derive(Debug, Deserialize)]
 #[serde(try_from = "RawConfig")]
 pub struct Config {
-    internal_url: String,
+    base_config: BaseConfig,
     phonenumber: String,
 
     dtmf_length: usize,
     result_length: usize,
-
-    decrypter: Box<dyn JweDecrypter>,
-    validator: Box<dyn JwsVerifier>,
 }
 
 // This tryfrom can be removed once try_from for fields lands in serde
 impl TryFrom<RawConfig> for Config {
     type Error = Error;
     fn try_from(config: RawConfig) -> Result<Config, Error> {
         Ok(Config {
-            internal_url: config.internal_url,
+            base_config: BaseConfig::try_from(config.raw_base_config)?,
```

Comment:
Since this is really only a wrapper for try_from from the common package, I think it would be better to not even use the double structure with try_from here, and just let serde handle the try from purely for the structure in the common package.

Reply:
BaseConfig is a class we control and have specified a good serde deserializer for. Because of that, serde can just immediately deserialize to BaseConfig, rendering the TryFrom and RawConfig here unneccesary